d2bs/kolbot/libs/common/Precast.js

```diff
@@ -152,18 +153,21 @@ var Precast = new function () {
 	};
 
 	this.precastSkill = function (skillId) {
-		var swapped,
-			slot = this.getBetterSlot(skillId);
+		var swapped;
 
-		if (slot !== me.weaponswitch) {
+		if (this.bestSlot[skillId] === undefined) {
+			this.bestSlot[skillId] = this.getBetterSlot(skillId);
+		}
+
+		if (this.bestSlot[skillId] !== me.weaponswitch) {
 			swapped = true;
 		}
 
-		this.weaponSwitch(slot);
+		this.weaponSwitch(this.bestSlot[skillId]);
 		Skill.cast(skillId, 0);
 
 		if (swapped) {
-			this.weaponSwitch(Math.abs(slot - 1));
+			this.weaponSwitch(Math.abs(this.bestSlot[skillId] - 1));
```

Comment:
change
`this.weaponSwitch(Math.abs(this.bestSlot[skillId] - 1));`
to
`this.weaponSwitch(this.bestSlot[skillId]^1);`